hack/build-and-push.sh

```diff
@@ -431,6 +434,34 @@ attach_migration_file() {
     return 0
 }
 
+# Find previous bundle that has a migration and output its digest.
+find_previous_migration_bundle_digest() {
+    local -r task_name=${1:?Missing task name}
+    local -r task_version=${2:?Missing task version}
+    local -r repo=${TEST_REPO_NAME:-task-${task_name}}
+
+    local -r url_list_recent_tags="https://quay.io/api/v1/repository/${QUAY_NAMESPACE}/${repo}/tag/?onlyActiveTags=true&limit=100"
+    local -r expr_filter_tags=".tags[] | select(.name | test(\"^(${task_name}-)?${task_version}-[0-9a-f]+\$\")) | .manifest_digest"
+
+    local prev_bundle_digest has_migration
+
+    curl --fail -sL "$url_list_recent_tags" | jq -r "$expr_filter_tags" | \
+        while read -r manifest_digest; do
+            curl --fail -sL -o /tmp/manifest.json \
+                "https://quay.io/v2/${QUAY_NAMESPACE}/${repo}/manifests/${manifest_digest}"
+            prev_bundle_digest=$(jq -r '.annotations."dev.konflux-ci.task/previous-migration-bundle"' </tmp/manifest.json)
+            has_migration=$(jq -r '.annotations."dev.konflux-ci.task.migration"' </tmp/manifest.json)
+            if [ "$has_migration" == "true" ]; then
+                echo "$manifest_digest"
+                return
+            fi
+            if [ "$prev_bundle_digest" == "null" ] || [ -z "$prev_bundle_digest" ]; then
+                echo
+                return
+            fi
```

Comment:
I'm not sure I fully understand the logic here. Why do we want to stop iterating through the manifest digests when we find one without the `dev.konflux-ci.task/previous-migration-bundle` annotation?

It would make more sense to me if the condition was backwards - if there's a non-empty `dev.konflux-ci.task/previous-migration-bundle` annotation, echo its value and return

Reply:
We stop because missing annotation means older task version without any migration. So it doesn't make sense to look at older versions - no migrations anyway.

Reply:
Hmm, in that case we probably don't need to iterate through the manifests at all. Wouldn't it be enough to look at the latest one?

* if it has a migration, return the manifest digest
* otherwise, return the value of the `dev.konflux-ci.task/previous-migration-bundle` annotation

Reply:
Ah, I see your other comment. +1